Adds proxy auth support to pulp_repo module

dcode opened the pull request:
##### SUMMARY
Added support for the pulp_repo module to allow for authentication to proxies. Adds the `proxy_username` and `proxy_password` options to the REST API calls.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
pulp_repo module

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/Users/<scrubbed>/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.4.0.0/libexec/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:53:22) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

dcode committed:
Add proxy auth to pulp_repo

sysadmind reviewed (changes requested):
Looks good. Just needs the pep8 updates that the Ansible commented.

dcode commented:
Working on it. I'll get it updated today.

On Thu, Nov 2, 2017 at 7:50 AM, Joe Adams <notifications@github.com> wrote:

> *@sysadmind* requested changes on this pull request.
>
> Looks good. Just needs the pep8 updates that the Ansible commented.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible/pull/32484#pullrequestreview-73774859>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKeNriEyKvLRCXOnFj5ddBh-9g5So2xks5sybqZgaJpZM4QPHP4>
> .
>

sysadmind commented on lib/ansible/modules/packaging/os/pulp_repo.py:
I think that `enable` indicates that this option can only enable the repoview and not delete it, but it looks like all of the code will still work to remove it as well (and it should). Looking at other modules, the phrasing is more often along the lines of `Whether to generate repoview files` or `Manage distributor repoview property`

dcode commented on lib/ansible/modules/packaging/os/pulp_repo.py:
Sounds good. Pushed a fix.

sysadmind reviewed (approved):
Looks good. Thanks for the contribution!

shipit

sysadmind commented:
shipit

dcode commented:
Sorry...I tested this in a sterile environment with a newer pulp installation and the `repoview` option failed via the API call without the `generate_sqlite` option. So, I added that option and now it works. The CLI tool automatically sets this, so using `repoview: True` will automatically set `generate_sqlite: True` in the module now.

resmo commented:
code LGTM
shipit

jsteinich committed:
add launchType to ecs_service per #35607

update ecs_taskdefinition to support fargate tasks per #35607

Camillevau committed:
Update docker_container.py

<!--- Your description here -->

 label: docsite_pr

dagwieers committed:
Add trailing dot and make it a separate item

mjmayer committed:
Add Fargate support for ECS modules

Fargate instances do not require memory and cpu descriptors. EC2 instances
 do require descriptions. https://botocore.readthedocs.io/en/latest/reference/services/ecs.html#ECS.Client.describe_task_definition

Fargate requires that cpu and memory be defined at task definition level.
EC2 launch requires them to be defined at the container level.

Fargate requires the use of awsvpc for the networking_mode. Also updated,
the documentation regarding where and when memory/cpu needs to the assigned.

The task_definition variable for the awspvc configuration colided with
the ecs_service for the bridge network. This would cause the test to fail.

Add testing for fargate

Add examples for fargate and ec2

bcoca committed:
doc fix

mattclay committed:
Implement new changelog generator.

bcoca committed:
ensure all config sources for ssh options

Akasurde committed:
Add note about updating pip (#41124)

Fixes: #39120

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

amenonsen committed:
Introduce inventory.any_unparsed_is_failed configuration setting (#41171)

In the process of building up the inventory by parsing each inventory
source with each available inventory plugin, there are three kinds of
possible errors (listed in order from earliest to latest):

1. One source could not be parsed by a particular plugin.
2. One source could not be parsed by any available plugin.
3. ALL sources could not be parsed by any available plugin.

The errors in (1) are a part of normal operation, e.g., the script
plugin is expected to fail to parse an ini-format source, and we will
ignore that error and try the next plugin. There is currently no way to
control this, and no known compelling use-case for a setting to control
it. This commit does not make any changes here.

We implement "any_unparsed_is_failed" to handle (2) above. If enabled,
this requires that every available source be parsed validly by at least
one plugin. In an inventory comprising a static hosts file and ec2.py,
this setting will cause a fatal error if ec2.py fails (a situation that
attracted only a warning earlier).

We clarify that the existing "unparsed_is_failed=true" setting causes a
fatal error only in (3) above, i.e., if NO inventory source could be
parsed. In other words, if there is ANY valid source in the inventory
(e.g., an ini-format static file), no combination of errors and the
setting will cause a fatal error.

If you want to execute your playbooks when your inventory is…

    (a) complete, use "any_unparsed_is_failed=true".
    (b) not empty, use "unparsed_is_failed=true".

The "unparsed_is_failed" setting should be renamed to
"all_unparsed_is_failed", but this commit does not do so.

Fixes #40512
Fixes #40996

Akasurde committed:
Correct defaults in lxc and lxd connection plugin

remote_addr value defaults to inventory_hostname, current defaults
seems to be copy-paste error.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
nxos_rpm module (#40849)

* Add nxos_rpm feature

* fix timing issues

* fix timing issues

* shippable fix

* review comment

* review comments fixes

* typo fix

ganeshrn committed:
Refactor ios cliconf plugin and ios_config module (#39695)

* Refactor ios cliconf plugin and ios_config module

*  Refactor ios cliconf plugin to support generic network_config module
*  Refactor ios_config module to work with cliconf api's
*  Enable command and response logging in cliconf pulgin
*  cliconf api documentation

* Fix unit test and other minor changes

* Doc update

* Fix CI failure

* Add default flag related changes

* Minor changes

* redact input command logging by default

Taluu committed:
composer: add new autoloader options (#40439)

* Add support for classmap-authoritative option for composer

* Add support for apcu-autoloader option for composer

* Fix a typo for the no-plugins option

Akasurde committed:
Minor typos in galaxy cli (#41083)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Do not consider an empty version string as a version (#41044)

When using an empty string as the version argument, the module would
before attempt to run something akin to:

    pip install module==""

This changes the behavior to:

    pip install module

Fixes #41043

mkrizek committed:
Warn when cond is used on meta tasks that don't support them (#41126)

achinthagunasekara committed:
Adding a switch to output docker logs (#40693)

* Adding a switch to print logs

* Update to docs

* Adding `version_added` for the new option

* Updating version added

* Moving `output_logs` before `paued` for alphabetically ordering.

SpamapS committed:
Enable check mode for docker_image_facts (#38944)

This module never changes anything. As such, there is no reason not to
just enable check mode.

tyll committed:
Add Fedora as possible ansible_distribution value (#41177)

"ansible_distribution" can also be Fedora

+label: docsite_pr

willthames committed:
Remove ECS policies from AWS compute policy

The compute policy was exceeding maximum size and contained
policies that already exist in ecs-policy.

Look up suitable AMIs rather than hardcode

We don't want to maintain multiple image IDs for multiple regions
so use ec2_ami_facts to set a suitable image ID

Improve exception handling

mkrizek committed:
Fix inserting noop tasks by using active state (#36056)

mkrizek committed:
first_found: fix examples to match the code (#40821)

Commit:
Update ios_facts.py (#40928)

Account for upper/lower case match occurrences of "[Nn]umber" and "[Ss]erial"

Model Number                       : WS-C3850-12X48U
System Serial Number          :  <removed>

dagwieers committed:
Add ACI and Meraki tests to resp. teams

Do the same as is done for NXOS

dagwieers committed:
Make ACI and Meraki module_utils supported by community

Just as is done for VMware module_utils

Qalthos committed:
Expand ios password prompt regex (#41131)

D3DeFi committed:
add zabbix namespace maintainer (#41181)

resmo committed:
cloudwatchevent_rule: doc: fix schedule_expression in example (#41138)

willthames committed:
[aws] Support custom KMS keys in `aws_s3` module (#35761)

* Allow the use of 'aws:kms' as an encryption method
* Allow the use of a non standard KMS key
* Deduce whether AWS Signature Version 4 is required rather than specifying with a parameter

dericcrago committed:
bump version of 'vcenter-test-container' to '1.3.0' (govmomi v0.18.0) (#41151)

* leave vmware_guest_powerstate tests enabled, but revert changes from: 66743f33
* leave VM poweroff tests enabled, but revert changes from: 87d6bdaf
* bumped 'vcenter-test-container' version to '1.3.0'
* updated test task names based on PR feedback

bcoca committed:
only add group if properly defined

also corrected skip message as this has to do with type, not name

fixes #40850

kbreit committed:
Meraki utility now loads() JSON (#40856)

* Changed request() to run json.loads() instead of module doing it
- Removed json.loads() from modules
- Removed some unreliable integration tests
- Removed self.function setting in construct_path()
-

* PEP8 changes

* Remove debug line for push

nitzmahone committed:
fix ansible-doc regression from missing plugins (#41167)

abhigarg28 committed:
[aws] Doc fix for aws_ses_identity (#41194)

Examples have a typo of ses_identity instead of aws_ses_identity. Fixed that and proposing these changes here.

+label: docsite_pr

s-hertel committed:
aws_s3: don't decrypt file before uploading - fixes #39287 (#39634)

* aws_s3: do not decrypt file before uploading to bucket

* changelog

quodlibetor committed:
Improve error message if policy_document does not exist (#40094)

jborean93 committed:
winrm: fix up unit tests (#41112)

LastCaress77 committed:
Update vmware_host_lockdown.py (#41189)

esxi_hostname should be 'list' instead of 'str'

Commit:
using lighter resource (#41223)

Nothing4You committed:
Add some dots at the end of sentences (#41176)

See e.g. https://docs.ansible.com/ansible/devel/reference_appendices/config.html#default-scp-if-ssh
In the docs the list of strings is just concatenated without additional interpunctuation.

+label: docsite_pr

jacum committed:
fixed passing non-default base_url from explicitly set cloud_environment (#40225)

trishnaguha committed:
increase timeout for connection reset by peer issue with httpapi (#41224)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

gdpak committed:
DCI failures (#41241)

justjais committed:
Metadata change to 'core' for 2.5 supported NIOS modules (#41244)

* changing metadata to core support

* changing metadata to core support

* changing metadata to core support

* changing metadata to core support

* changing metadata to core support

willthames committed:
aws_eks_cluster: New module for managing AWS EKS (#41183)

* aws_eks: New module for managing AWS EKS

aws_eks module is used for creating and removing EKS clusters.

Includes full test suite and updates to IAM policies to enable it.

* Clean up all security groups

* appease shippable

* Rename aws_eks module to aws_eks_cluster

ganeshrn committed:
Fix ios_user integration test failure (#41252)

* If the command input is dict from module
  in that case the check to see if command is
  end or `!` to exclude it from executing on remote
  host is wrong. Fix the logic to check `end` and
  `!` commands

sivel committed:
Revert #39365, improve error messaging (#41208)

* Revert " Update TaskInclude _raw_params with the expanded/templated path to file (#39365)"

This reverts commit 4b01b92cfefd2179d7b601a30e60a3fdf45fa35d.

* Improve error messaging, catch error templating parent path

agaffney committed:
Fix wrong python keyword (followup to #36357) (#41067)

Also add tests around that code path

abadger committed:
Make config files obey the utf-8 input rule

When we read files from disk we always want to read them as bytes and
then convert them to text ourselves.  This gives us greater control over
what encodings are used, what to do in case of errors decoding the bytes
to text, and better resilience against problems on both Python 2 and
Python 3.

If we left it up to Python to do this, on Python2, this could mean
that config values end up as bytes (leading to tracebacks elsewhere in
the code).  In Python3, it could traceback if the user's locale did not
match with the encoding of the ini file or config files could be decoded
as the user's locale encoding instead of as utf-8.

willthames committed:
Improve aws_s3 permission handling for non S3 (#38574)

* Test case for missing permissions

* Update aws_s3 module to latest standards

* Use AnsibleAWSModule
* Handle BotoCoreErrors properly
* Test for BotoCoreErrors
* Check for XNotImplemented exceptions (#38569)

* Don't prematurely fail if user does not have s3:GetObject permission

* Allow S3 drop-ins to ignore put_object_acl and put_bucket_acl

sivel committed:
Ensure the 2.7 porting guide is in the toc (#41267)

Akasurde committed:
VMware: update vmware_guest_tools_wait examples (#41246)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

cgroschupp committed:
[aws] Remove walrus conditional in aws_s3 module when using custom s3_url (#36832)

fix aws_s3 module to use custum s3_url.

s-hertel committed:
Fix async for aws_s3 - fixes #40281 (#40826)

* Fix async for aws_s3

* Add a test that async is able to be used on aws_s3 tasks

saito-hideki committed:
Add syslog_facility parameter handling with systemd.journal (#41078)

* Add syslog_facility parameter handling with systemd.journal

- Fixed issue #41072

Signed-off-by: Hideki Saito <saito@fgrep.org>

MironVeryanskiy committed:
Fix docs typo modules_intro.rst (#40835)

Fixed typo in modules_intro.rst.

+label: docsite_pr

rambleraptor committed:
GCP Inventory Plugin scopes fix

felixfontein committed:
Changing example code block language from yaml to yaml+jinja. (#40365)

bcoca committed:
no_log even when task_result doesn't provide key

 - now also checks task property
 - added reproducer to tests for unreachable status on item loop

nitzmahone committed:
fix truncated list functions in ansible-doc (#41281)

* fixed incomplete refactor of instance-level plugin list var

sivel committed:
minimal callback doesn't properly respect changed, bring into alignment with oneline (#41273)

whatupmiked committed:
Change falied_conditions to proper spelling (#41284)

samdoran committed:
Fix typeo in param name (#41287)

Akasurde committed:
ldap_attr: Use byte values instead of str (#41248)

The LDAP values may be of any kind (pictures, bytes, etc.) thus, ldap module enforce a "bytes" type.
We should pass properly encoded values instead of str

Fixes: #39569

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

justjais committed:
eos handles error gracefully when maximum sessions is reached (#41242)

* to resolve #36977

* to resolve #36977

* fix review comment

jainnikhil30 committed:
Update winrm.py (#41303)

fix the typos for winrm port

ganeshrn committed:
Fix junos_config integration test (#41304)

*  Change the ip address in fixture files
   as per PR #40554

Commit:
Changes to Yum Documentation (#41054)

* added an example for a list of packages to install and added C() to some areas since it cannot be read right
* fixed the changes to list for

rambleraptor committed:
GCP: Fix gcp_instance metadata keys (#41293)

sparunakian committed:
Add backup option to passwordstore lookup (and improve doc) (#39676)

* Add backup option in passwordstore lookup

* Update passwordstore lookup documentation

* Add precision regarding backup param

* Fix empty line added at EOF

* Add version_added attribute for backup option

* Switch examples to multi-line YAML

* Fix documentation for overwrite option

* Add changelog fragment for passwordstore lookup

* Update version added for new feature to 2.7


Co-authored by: Stéphane Parunakian <stephane.parunakian@smile.fr>

tbouvet committed:
Add new module to create/update a docker swarm. (#38280)

* Add new module to create/update a docker swarm.

* Fix ansible-test sanity

* Fix requirements

* Fix requirements

* Add tag for author

* Test integration test.

* Fix main.yml

* Add linux arch

* Add template

* Fix test result

* Integration test to create/remove a swarm manager

* fix join test

* Downgrade docker-py

* fix rhel

* Fix review documentation.

* Fix whitespace

* Check docker installation.

* test docker install

* check

* Remove docker socket

* Fix docker install

* Fix sanity test

* Rebase

* Add docker_swarm maintainer

* Fix review

* Fix new version.

* Add docker default values

* Fix description.

* Reworked documentation

* Fix YAML error

* Rebase

* Fix example for update state.

* Fix idempotent states. Fix states: present/absent.

* Fix sanity

* Fix variables sanity

* Update example for absent state.

* fix sanity

* Wrap the contents of error message in to_native.


Co-authored by: Dag Wieers <dag@wieers.com>

smolz committed:
add example using peer keepalive vrf and delay restore (#41111)

* add example using peer keepalive vrf and delay restore

<!--- Your description here -->
add example using peer keepalive vrf and delay restore
+label: docsite_pr
+label: issue ansible/community#311

* Update nxos_vpc.py

update task name to include "existing"

permitanyany committed:
Added show interface example in JSON format (#40732)

* Added show interface example in JSON format

Added show interface example in JSON format

+label: docsite_pr
+label: issue#311

* fixed whitespace

wazzaz committed:
Format fix on VMware Guide (#41272)

* Updating format of VMware Guide: small formatting error - hyperlink to module

sivel committed:
Don't ignore a duplicate host for an already processed include (#40361)

* Don't ignore a duplicate host for an already processed include, assume that the repetition indicates a new include. Fixes #40317

* Add intg tests to ensure duplicate items in loop are not deduped

* Add note about relative indexing

mattclay committed:
Update 2.6 roadmap schedule.

toshipp committed:
Fix a process execution using lxc connection plugin on Python3.

python2-lxc module needs bytes, on the other hand python3-lxc requires text.
To solve such incompatibility, use to_native other than to_bytes.
This fixes #41060.

Akasurde committed:
VMware: Allow user to select disk_mode (#38951)

* VMware: Allow user to select disk_mode

This fix allows user to select disk modes for given disk configuration
in the given VM.

Fixes: #37749

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

* Review comments

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

stoned committed:
ini lookup plugin: fix documentation layout (#41374)

trishnaguha committed:
Fix nxos_snapshot compare (#41386)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

Akasurde committed:
Convert valid_date to bytes for conversion (#40872)

Fixes: #40523

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

tmshn committed:
BSDTimezone: distinguish UTC and Etc/UTC (#41234)

* Fixed BSDTimezone to distinguish UTC and Etc/UTC

* Added test for timezone to disinguish UTC vs Etc/UTC

remyleone committed:
Remove VC scaleway (#40791)

maxamillion committed:
Revert "Remove VC scaleway (#40791)" (#41401)

This reverts commit 007127ea27b7dcadba8f297ccb953d42e3d3d22d.

maxamillion committed:
Revert "Revert "Remove VC scaleway (#40791)" (#41401)" (#41402)

This reverts commit 391509006ce73c707753fac29b6b494b6f6cba72.

samdoran committed:
Update docs on import/include tasks (#33032)

sivel committed:
Add docs for debugger for keyword dumper (#41421)

pilou- committed:
Default squash actions add pip (#41390)

* pip tests: remove trailing spaces

* pip tests: use Jinja tests

* fixup! pip tests: remove trailing spaces

* pip tests: use 'command' instead of 'shell' module

* pip tests: remove unused variable

* pip tests: use a package with fewer dependencies

sampleproject has one dependency: 'peppercorn' and peppercorn doesn't
have any dependency.

* pip tests: check that 'name' param handles list

* pip: squash package parameters

Note that squashing will be removed in 2.11, new code should directly
use a list with the 'name' parameter.

Commit:
more integration test improvements (#41184)

Commit:
fix get existing vpc issue (#41429)

wisotzky committed:
fix netconf_get documentation section and lock logic (#40357)

* fix netconf_get documentation section and lock logic

* updated lock behavior

BondAnthony committed:
Digital Ocean Block Storage Bug (#41431)

kaneda-fr committed:
vca_vapp: Connecting app network to vApp and VM after vApp creation (#19886)

* Connecting app network to vApp
This fixes connection mechanism for connecting app network to vApp and VM
after vApp creation, returning the IP assigned to the VM as fact 'vm_ip'
* removed duplicate check on task fail for vApp creation. Fixed issue with network_mode value for VCD version > 5.7
* fixed trailing white space

Signed-off-by: Kaneda-fr <sebastien@lacoste-seris.net>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sivel committed:
When using ANSIBLE_JINJA2_NATIVE bypass our None filtering in _finalze (#41408)

* When using ANSIBLE_JINJA2_NATIVE bypass our None filtering in _finalize. Fixes #41392

* Add tests for _finalize bypass

* Address python3 failures in tests

mscherer committed:
Return changed=True if a new certificate was written (#41445)

abadger committed:
Add changelog entry for the no_log fix

hellt committed:
fix handling of ># ending prompts for sros (#41382)

s-hertel committed:
Add AWS boto3 error code exception function is_boto3_error_code (#41202)

* Add aws/core.py function to check for specific AWS error codes

* Use sys.exc_info to get exception object if it isn't passed in

* Allow catching exceptions with is_boto3_error_code

* Replace from_code with is_boto3_error_code

* Return a type that will never be raised to support stricter type comparisons in Python 3+

* Use is_boto3_error_code in aws_eks_cluster

* Add duplicate-except to ignores when using is_boto3_error_code

* Add is_boto3_error_code to module development guideline docs

abadger committed:
Tasks that are expected to fail need to begin with a special string

nitzmahone committed:
allow dev and prerelease at the same time (#41464)

mattclay committed:
Disable failing acme_account integration test.

Akasurde committed:
plugins: Doc improvement (#41378)

* Typo fixes
* Grammer fixes
* Option fixes

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
reappied changes in a new pr (#41228)

sivel committed:
Ensure we raise the exception caught _post_validate_environment->_parse_env_kv. Fixes #41322 (#41411)

sdodsley committed:
Fix race condifiton where multiple hosts can try and create or delete (#39698)

the same volume, snapshot or hostgroup,

jaevans committed:
Add support for Tower Smart inventories (#41458)

* Support Smart Inventories

Add kind and host_filter fields and pass through to tower_cli.

* Add documentation for new Smart Inventories options

* Add missing description header for host_filter documentation

* Add version added tags to new options

* Bumped vesion_added to 2.7

maxamillion committed:
service_facts correct meaning of state for systemd service units (#40914)

* service_facts correct meaning of state for systemd service units

Fixes #40809

Previously this module used the commend `systemctl list-unit-files
--type=service` to query state of services but list-unit-files only
shows enabled vs disabled which is not what we want for "state"

Signed-off-by: Adam Miller <admiller@redhat.com>

* make sure to define service_name before referencing it

Signed-off-by: Adam Miller <admiller@redhat.com>

bcoca committed:
corrected version

plumbeo committed:
Add nomask option to the acl module (#39810)

* Add nomask option to the acl module

* acl: allow to choose if and when to recalculate the effective right masks of the files

bengerman13 committed:
expand user before calling abspath (#40196)

* expand user before calling abspath

* omit expanduser code smell check for volumes

Commit:
fix reload smu issue (#41270)

* fix reload smu issue

* review comments

justbiaoliu committed:
Add cli and netconf ability from ansible itself for cloudengine ce modules (#41357)

* for shippable

for shippable

* add cliconf

* add network_cli

* add cliconf and network cli and netconf

* modify bugs of netconf

* add shippable modify

* update shippable

update shippable

geerlingguy committed:
Fix typos in blocks documentation examples. (#41508)

trishnaguha committed:
don't retrieve config in running_config when config is provided for diff (#41400)

* don't retrieve config in running_config when config is provided for diff

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* fix for eos, nxos

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* add integration test

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

acalm committed:
accept + and - modifiers for file attributes (#40061)

* accept + and - modifiers for file attributes
* tests for adding/removing file attributes

abadger committed:
Add changelog fragment for +/- attribute handling

Qalthos committed:
Check for ConnectionError on change of config or commands (#41504)

Im0 committed:
Adding success_msg to assert (#41048)

* Adding success_msg to assert

* Added missing version_added. Importing and checking string_types.

trishnaguha committed:
fix nxos edit_config for httpapi and have uniform load_config (#41358)

* fix nxos load_config for httpapi and migrate to cliconf

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* add comment

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* address review comment

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* address review comment

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

Akasurde committed:
Remove extra line in file_attributes.yaml

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Rename argument msg to fail msg

This change make the arguments of assert less ambiguous.

jsimpso committed:
Fixes #35622 (#41522)

schp committed:
sort the port list properly (.sort() returns None)

Commit:
Point out that fail_msg is new in 2.7

djzager committed:
Update APB templates (#41559)

felixonmars committed:
Fix some typos in config/base.yml (#41363)

Akasurde committed:
Correct typo from 'Valut' to 'Vault' (#41574)

Correct typo from 'Valut' to 'Vault'

+label: docsite_pr

BondAnthony committed:
DigitalOcean - Fix TypeError when users have tags named the same as droplets (#41368)

* Fix TypeError when users have tags named the same as droplets
* Updated code based on review comments

ivica-k committed:
setting LANG, LC_ALL and LC_NUMERIC to C (#41544)

* This forces the fact gathering commands to use C locale, preventing
weird locale issues such as different separator etc

Akasurde committed:
VMware: Add documentation for customValue (#41399)

* VMware: Add documentation for customValue

In order to populate custom values (custom attributes of virtual machine)
inside hostvars, customValue needs to be uncommented under the section
properties in vmware_inventory.ini

* Remove unwanted single quotes from custom_field_group_prefix

Fixes: #41395

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Add documentation around package dependencies (#41437)

Documented procedure to handle package dependencies required by
packages which are Ansible dependencies.

Fixes: #30732
Fixes: #16571

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Add more info related to galaxy sub commands (#41085)

Fixes: #17469

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sharph committed:
docker: Allow publishing of ports with the same number but different protocol (#38412)

* Don't deduplicate docker container ports with different protocols

* Test _parse_exposed_ports

nitzmahone committed:
skip winrm unit tests if winrm is not installed (#41596)

abadger committed:
Add changelog for gce_net sort fix

https://github.com/ansible/ansible/pull/41567

samdoran committed:
Update user module docs (#41572)

* Clarify append and groups behavior
* Document return values

abadger committed:
Temporarily revert c119d54

There were bugs in this that needed to be resolved.  No time to get the
fix reviewed sufficiently for 2.6.0.

We'll get this into 2.7.0 and try to get this into 2.6.1 as well.

Will need the work done in https://github.com/ansible/ansible/pull/36218
when it does get merged.

caphrim007 committed:
Fixes usage of popen in bigip iapplx package (#41612)

This functionality is superceeded by the run_command method in the
ansible module class.

kbreit committed:
Add docstrings to meraki module utility methods

PR #41369

Co-authored-by: Kevin Breit <kevin.breit@kevinbreit.net>

Commit:
fix nxos_linkagg issue (#41550)

* fix nxos_linkagg issue

* fix shippable

jdufresne committed:
Update additional pypi.python.org URLs to pypi.org (#41373)

For details on the new PyPI, see the blog post:

https://pythoninsider.blogspot.ca/2018/04/new-pypi-launched-legacy-pypi-shutting.html

Akasurde committed:
Fix typo in group_by module (#41627)

Fixes: #31600

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

s-hertel committed:
read scaleway option after the options have been set (#41259)

jdufresne committed:
Prefer readthedocs.io instead of readthedocs.org for doc links (#41537)

Read the Docs moved hosting to readthedocs.io instead of
readthedocs.org. Fix all links in the project.

For additional details, see:

https://blog.readthedocs.com/securing-subdomains/

> Starting today, Read the Docs will start hosting projects from
> subdomains on the domain readthedocs.io, instead of on
> readthedocs.org. This change addresses some security concerns around
> site cookies while hosting user generated data on the same domain as
> our dashboard.

hwoarang committed:
facts: suse: Fix major version detection for SUSE (#41416)

When parsing the distribution files such as /etc/os-release, we extract
the full distribution version but not the major version. As such, the
ansible_distribution_major_version ends up being 'NA' whereas the
ansible_distribution_version contains the full version.

Before this patch we get this on openSUSE Leap 15

ansible -o localhost -m setup -a filter=ansible_distribution_major_version
localhost | SUCCESS => {"ansible_facts": {"ansible_distribution_major_version": "NA"}, "changed": false}

After this patch we get this

ansible -o localhost -m setup -a filter=ansible_distribution_major_version
localhost | SUCCESS => {"ansible_facts": {"ansible_distribution_major_version": "15"}, "changed": false}

This also fixes the Tumbleweed distribution test to report a proper
major version and also adds a test for openSUSE Leap 15.0 to avoid
potential future regressions.

Fixes: #41410

zharalim committed:
Update playbooks_filters.rst (#41628)

Add an example on how to use regexp_replace to modify list items.

+label: docsite_pr

Fix typo

Commit:
Fix vmware host config for all vmware OptionTypes (#41343)

* Fix vmware host config for all vmware OptionTypes

Setting some options and option types failed with invalid value errors being return from vmware, this resolves all known ways that issue can occur.

* Add logic for integer inputs as string

For example - "UserVars.ESXiShellInteractiveTimeOut": "20"

Fixes #40180
Fixes #41212

jamescassell committed:
update play_context for reset_connection (#40866)

steps taken from lib/ansible/executor/task_executor.py

fixes: https://github.com/ansible/ansible/issues/27520

sivel committed:
Load role vars and defaults before parsing tasks (#40982)

* Load role vars and defaults before parsing tasks. Fixes #40163

* Add porting guide note

* Wording clarifications

* typo

* grammar fixes

pilou- committed:
Scaleway inventory plugin: small improvements (#41642)

* scaleway inventory: token is mandatory

* scaleway inventory: mention exception in error

* scaleway inventory: remove print statement

* scaleway inventory: options don't need to be attr

* scaleway inventory: remove unused attr

Commit:
Check type of fail_msg and success_msg

Commit:
More explanation in docstring

Commit:
Improve code style

agaffney committed:
Fix name of action plugin in error message (#41665)

Commit:
Update my own author email in comments block

my ansible/redhat email is not valid, replacing it with something that is.

emonty committed:
Fix openstack inventory script for multi-cloud case (#41664)

The shift to openstacksdk left us with a bug in that when running bare
with --list, the cloud argument to get_one is None. We just need _one_
of the clouds to pull the cache settings, since they are global (yet,
we'll go back and fix this in sdk)

If it's None, just use get_all and grab the first one.

jasontedor committed:
Enable to use greedy checks for outdated casks (#40799)

* Enable to use greedy checks for outdated casks

When using brew cask outdated to check if an installed cask is outdated
or not, brew cask will skip casks that have auto_updates set to true or
version: latest. This means that Ansible tasks using the homebrew_cask
module to upgrade packages installed by brew cask will miss upgrading
such packages. However such packages can still be managed by brew cask
so we need to be able detect such packages. This can be done with the
--greedy flag passed to brew cask outdated as this will also include
such packages that are outdated. This commit adds a greedy parameter to
the homebrew_cask module to enable upgrading such packages using Ansible
tasks with the homebrew_cask module. The default behavior preserves the
same behavior as today. Example usage would be:

- homebrew_cask:
  name: 1password
  state: upgraded
  update_homebrew: yes
  greedy: yes

* Fix test issues

* Add extra comma to match style

caphrim007 committed:
Fixes the default assignment of values (#41672)

Fixes: #41640

The defaults should be inherited from the parent during initial
creation. This patch fixes that.

Commit:
[docs] Include "_facts" in module examples section (#41680)

The example says module name: "cloudwatchlogs_log_group"
where it should say: "cloudwatchlogs_log_group_facts"

+label: docsite_pr

Deepakkothandan committed:
[ecs_service] fix assign_public ip for network configuration (#41685)

* fix assign_public ip for ecs_service

* used module.botocore_at_least instead of distutils

ryansb committed:
Fix aws_config_aggregator argument count error (#41686)

NilashishC committed:
Fix vyos_vlan aggregate issue & added tests (#41638)

* Fix vyos_vlan aggregate issue & Added tests

* Fix #2 for vyos_vlan aggregate issue

maxamillion committed:
add notes to service_facts about accessing fact data (#41608)

* add notes to service_facts about accessing fact data

Signed-off-by: Adam Miller <admiller@redhat.com>

* remove unquoted :

Signed-off-by: Adam Miller <admiller@redhat.com>

ryansb committed:
Fix exception message in AWS lambda module

bcoca committed:
added missing raise

ryansb committed:
[aws] Remove double-defined variable from iam_policy module

jyundt committed:
Add upgrade ('latest') support to pkgng module (#41623)

* Add upgrade ('latest') support to pkgng module

* fixup for bad yaml

* fixup for bcoca review:
- add `version_added:` to description
- change examples to use `name:` and remove comments
- DRY out install_packages()
- clean up state conditional check in modules.params

* fixup to remove extra whitespace

* fixup to handle custom pkgsite when checking for updates

* fixup to remove misleading `version_added:`

Akasurde committed:
Raise Exception in virtualbox inventory plugin

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Add RETURN docs for the pip module (#39907)

The pip module returns some things. Let's document them.

LindsayHill committed:
Fixed SLX-OS docstring inconsistency (#41662)

ryansb committed:
[aws] Simplify conditional in iam_cert module (#41691)

ryansb committed:
[aws] lambda_policy: Remove spurious definition of `policy` variable (#41693)

ryansb committed:
[aws] remove unused variable in iam_user module (#41695)

s-hertel committed:
[ecs_service] fix assign_public_ip network configuration to module parameter rather than fixed string (#41689)

Fix parameter name

Use suboptions

document suboptions

Add a test to assert assign_public_ip is configurable

ryansb committed:
[aws] Fix unused return value in ec2_vpc_vgw_facts module (#41697)

ryansb committed:
[aws] Remove unused variable in ec2_asg module (#41706)

ryansb committed:
[aws] Remove extra variable definition in ec2_vpc_nacl module (#41707)

ryansb committed:
[aws] Remove unused return in ec2_vpc_igw_facts module (#41708)

ryansb committed:
[aws] Remove unused return in ec2_vpc_dhcp_option_facts module (#41709)

mattclay committed:
Temporarily disable gem test on osx.

Disabled due to lack of newer TLS version support on the test system.

Akasurde committed:
VMware: Add timeout in wait_for_task (#39596)

* VMware: Add timeout in wait_for_task
* Added exponential backoff algorithm
* code review comments

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
openssl_csr: Update example (#41677)

Update example of Subject Alternative Name with dynamic list.

Fixes: #33676

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

kbreit committed:
Add proper response documentation (#41370)

- Added response documentation for organizations
- Additional information is likely to be added later

kbreit committed:
Removed redundant documentation (#41422)

- Removed response documentation on keys which are standard across
modules.

kbreit committed:
Meraki_snmp - Add examples for SNMP documentation (#41424)

* Add examples for SNMP documentation

* Removed whitespace for PEP8

kbreit committed:
Added support for passing org_id instead of org_name (#41518)

- Faster execution if passed instead of org_name
- Updated documentation
- Added additional integration tests

kbreit committed:
Improved support for parameter testing and checks for org_id (#41517)

kbreit committed:
Remove notes since it's already specified in the doc fragment (#41614)

kbreit committed:
Improve support for org_id in module (#41519)

- Faster performance since it reduces number of API calls
- Also simplified some of the execution

kbreit committed:
New module - meraki_device (#41631)

* Initial commit for meraki_device module
- Allow claiming, removal, updating, and querying of devices
- Integration tests are included
- Integration tests are not complete because physical gear is required
- Integration tests also require Meraki subscriptions

* Added support for serial number query without network

* Added support for net_id and net_name

* Changes recommended by ansible-test for PEP8 and documentation

* Remove duplicate state in example

* Fix typo

kbreit committed:
New module - meraki_vlan (#41632)

* Initial commit of meraki_vlan module
- Create, delete, modify, and query VLANs within a network
- Support for all allowed objects in the VLAN data structure
- Meraki defaults networks to have VLANs disabled and there is no
way to use the API to enable VLAN support. It must be enabled
manually.

* Fix formatting error in documentation

* Formatting changes and added documentation

* PEP8 fix

kbreit committed:
New module - meraki_config_template (#41633)

* Implement configuration template management
- Queries or removes templates
- Can bind or unbind templates to networks
- Module is idempotent only for binding and unbinding
- Meraki does not allow template creation via API
- Integration test is tedious b/c previous bullet point
- Fixed bug in construct_path() so it won't set self.function

* PEP8 changes

* Re-enable some integration tests, use variables, and fix broken code

kbreit committed:
New module - meraki_switchport (#41634)

* Initial commit for meraki_switchport module
- Query or modify swichport configurations
- Further optimizations are available
- Integration tests will require manual editing of file for others

* Remove blank lines

mkrizek committed:
apt: fix apt-mark on debian6 (#41530)

gdpak committed:
net_put module: add default mode to copy binary files (#41476)

* add default mode to copy binary files

* sanity fix

* add version for new options

bcoca committed:
handle groups correctly when they are 'null' (#41698)

* handle groups correctly when they are 'null'

 - even if defined as mapping but having no keys, objects shoudl still be processed correctly
 - also add ansilbe_verbosity to vars not to display in ansible-inventory

fixes #41692

justjais committed:
To fix eos_vrf failure when transport method is eapi (#41470)

* resolve bug 40930

* resolve bug 40930

* to fix review comments

* to fix review comments

* reverting the changes based on review

Akasurde committed:
Argument spec must be dict/hash (#40858)

validate-modules should fail when argument is not dict/hash.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

evgeni committed:
lineinfile: add regex as an alias for regexp (#20454)

the param is called regexp, but a lot of people spell it regex
add an alias to help them

maxamillion committed:
ensure copy action plugin returns an invocation in result (#41426)

* ensure copy action plugin returns an invocation in result

Fixes #18232

Previously the action plugin for copy, which runs operations on the
control host to orchestrate executing both local actions on the
control host and remote actions on the remote host, is not returning
invocation dict in the result dict, this happens here where the
return from _copy_file() is None

When force is True, the _execute_module() method is called, which
returns the dict containing the invocation key. This patch ensures
there is always an invocation key.

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix conditional, handle content no_log

Signed-off-by: Adam Miller <admiller@redhat.com>

caphrim007 committed:
Marks select modules as stableinterface (#41741)

this is being required by customers so that they may adopt ansible
and F5's work in ansible.

abadger committed:
Fix file state=touch not returning diff information

Fixes #41755

jborean93 committed:
win_updates fix when win_updates is run with async (#41756)

Akasurde committed:
Remove redundant API method from foreman inventory (#41720)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

abadger committed:
Remove an unnecessary import from the ansiballz wrapper (#41747)

s-hertel committed:
ecs_service: make assign_public_ip option a boolean (#41759)

psykotox committed:
new provider: ownca (#35840)

tacid committed:
Update zabbix_screen.py limit screen columns (#41487)

* Update zabbix_screen.py limit screen columns

Add option graphs_in_row that allows to limit the count of columns on zabbix screen. When graphs_in_row columns is filled, begins to draw next row. If there is many hosts with many graphs each, draws all graphs for one host in an column each below each other

* fix remove whitespaces before and after brackets

Fixing ansible-test sanity --test pep8 errors
moved

* fix documentation parsing error

Fixing error: missing documentation (or could not parse documentation): expected string or buffer

willthames committed:
Trim parameter documentation when generating docs (#38470)

While the HTML produced is perfectly valid, without the `trim` filter,
a lot of warnings are emitted (700 lines of warnings out of 2812 are
eliminated by this change)

mkrizek committed:
file: clarify attributes usage in docs (#41780)

Commit:
[docs] Change 'tag' to 'tags' in aws_ec2 inventory examples (#41438)

Shaps committed:
Add shell out checks (#41545)

* Added error codes for shell_out checks

* Added ignore lines for allowed Modules

* Added shell out checks

* Fixed pep8

* Updated regex to only match subprocess.Popen

* Added failing modules to ignore.txt

* Wrong postgresql module in ignore.txt

* Removed bigip from ignore.txt

cpollard0 committed:
[aws][docs] Update max session duration in sts_assume_role documentation(#41765)

Update to reflect assume role updated durations as of March 2018. 
https://aws.amazon.com/blogs/security/enable-federated-api-access-to-your-aws-resources-for-up-to-12-hours-using-iam-roles/

+label: docsite_pr

s-hertel committed:
Document compose for aws_ec2 inventory plugin (#40621)

* Document compose for aws_ec2 inventory plugin

* split up examples into several parts with some additional comments

machacekondra committed:
ovirt_auth: In case of token is passed u/p is not requierd (#41788)

ryansb committed:
Remove unused imports from aws_config_aggregator (#41803)

Akasurde committed:
FreeIPA: Add support for TTL in dnsrecord (#41768)

Fixes: #33969

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ryansb committed:
Remove unused imports from aws_direct_connect_connection module (#41802)

* Remove unused imports from aws_direct_connect_connection module

* Fix tuple notation

jmighion committed:
Revert "Fixes #35622 (#41522)" (#41805)

This reverts commit 8357ae69e71081a6eabada733974204a11efd225.

mattclay committed:
Disable failing rabbitmq_plugin test.

rambleraptor committed:
[gcp] Fix service account email parameter on gcp_compute_instance (#41806)

bcoca committed:
fix minor issues with debug and item labels (#41331)

* fix minor issues with debug and item labels

 - no more `item=None`, we always have a label now
 - debug should only show expected information, either msg= or the var in var=
 - also fixed method name, deprecated misleading _get_item

mattclay committed:
Disable remaining failing rabbitmq_* tests.

obourdon committed:
Add random_mac string filter (#39775)

Add new filter to generate random MAC addresses from
string prefix. See docs/docsite/rst/user_guide/playbooks_filters.rst
for more detailed infos.

bcoca committed:
fix vars hostname fallback

also made it optimistic, rely on exceptions instead of copmlex if chains

bcoca committed:
exclude lookup_terms from config errors (#41740)

* exclude lookup_terms from config errors
* moved direct

abadger committed:
Bump versions on release_and_maintenance doc

(cherry picked from commit c8b1b271c66ff249bfd590e789fb1941c9887d9f)

kbreit committed:
Remove extarneous json.loads() which was moved to the utility. (#41731)

Fred-sun committed:
Update azure_rm_publicipaddress_facts.py example (#41833)

Example describe error

maxamillion committed:
validate vars_files entries are string_type or list (#40985)

* validate vars_files entries are string_type or list

Fixes #17594

Signed-off-by: Adam Miller <admiller@redhat.com>

* use collections.Sequence to simplify if check

Signed-off-by: Adam Miller <admiller@redhat.com>

lihongjie0209 committed:
update ansible.cfg example (#41832)

sijis committed:
Fix: wildcard excludes in unarchive with tar archives (#40935)

* fix: exclude using wildcards for tar archives

Fixes #37842, #22947

* fix: Remove quote() as it munges the exclude format

* test: Refactor to use single archive structure

A common structure archived by different methods should simplify some of
the feature tests.

* test: Use common archive layout to validate exclude feature

* test: Use the same exclude checks for zip/tar archives

sdodsley committed:
Fix API call to get ActiveCluster information and add (#41744)

connected arrays and volume group counts to default

Qalthos committed:
Netconf bytes fixes (#41607)

* Solve some bytes issues on iosxr

* Solve some bytes issues in junos

* Do the correct thing with tostring based on lxml or not

permitanyany committed:
Show BGP Status In JSON (#41844)

Qalthos committed:
Make validate_certs togglable for httpapi (#41538)

* Make validate_certs togglable for httpapi

* Add `version_added`

sivel committed:
Handle typing of vars_files in Play.get_vars_files. Fixes #14708 (#41801)

sivel committed:
Validate that rescue and always have an accompanying block. Fixes #26751 (#41822)

permitanyany committed:
Add 2 EBGP neighbors with template (#41853)

<!--- Your description here -->

+label: docsite_pr

dpassante committed:
cloudstack: new module cs_disk_offering (#41795)

gthieleb commented:
This commit fixes the authenticaton for `pulp_repo` module against cooperate proxy and works in our environment with RHEL7 and pulp 2.14.0.

Please ship that fix with current ansible stable.

```
$ rpm -qa|grep pulp|grep rpm
pulp-rpm-admin-extensions-2.14.0-1.el7.noarch
pulp-rpm-plugins-2.14.0-1.el7.noarch
python-pulp-rpm-common-2.14.0-1.el7.noarch
pulp-rpm-consumer-extensions-2.14.0-1.el7.noarch
pulp-rpm-handlers-2.14.0-1.el7.noarch
pulp-rpm-yumplugins-2.14.0-1.el7.noarch
```

jborean93 committed:
win_chocolatey: always return the rc return value (#41883)

jborean93 committed:
windows dsc docs: fix missing colon in example (#41886)

agaffney committed:
Merge various stdout callback plugins into 'default' (#41058)

This allows mixing and matching of stdout callback features

ganeshrn committed:
vyos and ios cliconf plugin refactor (#41846)

* vyos and ios cliconf plugin refactor

*  Refactor vyos cliconf plugin
*  Change vyos module_utils and vyos_config as per refactor
*  Minor changes in ios cliconf plugin

* Fix unit test failure

* Fix sanity issues

* Add get_diff to rpc list

felixfontein committed:
New module: acme_certificate_revoke (#40995)

* First version (without revocation checking).

* Adding check mode and OCSP revocation verification.

* Fixing ACME v1 behavior.

* Fixed superfluous space.

* Fixing links.

* Working around linter complaints.

* Added docstring.

* More defensive.

* Disabling check mode for now.

* Simplifying module by no longer checking OCSP, and removing check mode vestiges.

paneu committed:
cliconf: Fix use of multiple prompts in send_command (#41885)

Upon preparing the commands for sending to the device, cliconf converts
the optional prompt to a byte string. However, since there might be
multiple prompts specified, the conversion has to happen for each prompt
individually. Otherwise, wrong regexes will be compiled in
_handle_prompt from network_cli Connection.

ganeshrn committed:
Fix ios and vyos cliconf edit_config return type (#41896)

* Fix ios and vyos cliconf edit_config return type

Modify cliconf edit_config api to return a json string with
diff and response received from remote host for ios and vyos.

* Doc change

kbreit committed:
Bugfix for idempotency bug (#41889)

- Comparison was not happening properly as it lacked full data
- Module now creates a full data structure on payload
- Defaults to America/Los_Angeles as that's what Meraki seems to do

permitanyany committed:
Add Show Interface Example (#41863)

<!--- Your description here -->

+label: docsite_pr

dcode commented:
I was trying to rebase this, but clearly messed up. Working on fixing it. I might open a new PR with my squashed changes if I can figure it out.
